### ISSUE TITLE
support google's inception model

### ISSUE THREAD

**Issue by dribnet** [jcjohnson/neural-style#43]
Opening the idea of adding support for google's [pre-trained network for the Inception model](https://github.com/google/inception).

Pros:
  * more accurate than VGG_ILSVRC_19
  * more permissive license (VGG_ILSVRC_19 is [non-commercial use only](https://gist.github.com/ksimonyan/3785162f95cd2d5fee77))

Cons:
  * Caffe2 model looks non-trivial to port. Data is stored in the order height-width-channel instead of channel-height-width, the padding convention is slightly different, etc.

**Comment by jcjohnson** [comment]
I agree that this would be awesome, but it seems really tricky to get working. It's certainly not supported by loadcaffe, so it would have to be a custom solution just for the GoogLeNet network. As you mention we'd have to double-check to make sure that the same padding / pooling / rounding conventions are used between caffe2 and torch nn layers, which would be annoying.

On the other hand, GoogLeNet is much faster and uses much less memory than VGG-19, both of which would be great.

**Comment by danielvarga** [comment]
https://github.com/kaishengtai/neuralart can switch between these two models. Is kaishengtai's solution non-applicable here? In case it is relevant, note that both projects use the MIT license.

**Comment by spoid** [comment]
I am confused, on the VGG website http://www.robots.ox.ac.uk/~vgg/research/very_deep/ they link to a difference Creative Commons license that allows commercial use.

**Comment by 3DTOPO** [comment]
They kindly changed it to CC Commercial after I contacted the authors.

**Comment by htoyryla** [comment]
There still is a contradiction concerning the license on VGG16 and VGG19. On their page http://www.robots.ox.ac.uk/~vgg/research/very_deep/ they say "The models are released under Creative Commons Attribution License" but the information files linked a few lines below still refer to Creative Commons Attribution Non-commercial.

This is somewhat confusing.

**Comment by htoyryla** [comment]
Just noticed that the ambiguity concerning the VGG licenses has been removed: the actual info pages now point to http://www.robots.ox.ac.uk/~vgg/research/very_deep/  where the license is CC attribution.
